includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1196,7 +1203,12 @@ public function maybe_process_upe_redirect() {
 			return;
 		}
 
-		$this->process_upe_redirect_payment( $order_id, $intent_id, $save_payment_method );
+		$redirect_on_error = '';
+		if ( isset( $_GET['redirect_on_error'] ) && self::REDIRECT_ON_ERROR_PAY_FOR_ORDER === $_GET['redirect_on_error'] ) {
```

Comment:
(optional) Since this is the only possible option, we could also treat this as a boolean (like `redirect_to_pay_for_order_on_error`). But this does not make sense if we have any indication that other URLs will be supported in the future.